src/main/java/org/opensearch/flowframework/workflow/CreateIndexStep.java

```diff
@@ -96,15 +119,132 @@ public String getName() {
+                        if (r) {
+                            // return true if update index is needed
+                            client.admin()
+                                .indices()
+                                .putMapping(
+                                    new PutMappingRequest().indices(indexName).source(mapping, XContentType.JSON),
+                                    ActionListener.wrap(response -> {
+                                        if (response.isAcknowledged()) {
+                                            UpdateSettingsRequest updateSettingRequest = new UpdateSettingsRequest();
+                                            updateSettingRequest.indices(indexName).settings(indexSettings);
+                                            client.admin()
+                                                .indices()
+                                                .updateSettings(updateSettingRequest, ActionListener.wrap(updateResponse -> {
+                                                    if (response.isAcknowledged()) {
+                                                        indexMappingUpdated.get(indexName).set(true);
+                                                        internalListener.onResponse(true);
+                                                    } else {
+                                                        internalListener.onFailure(
+                                                            new FlowFrameworkException(
+                                                                "Failed to update index setting for: " + indexName,
+                                                                Response.Status.INTERNAL_SERVER_ERROR
+                                                            )
+                                                        );
+                                                    }
+                                                }, exception -> {
+                                                    logger.error("Failed to update index setting for: " + indexName, exception);
+                                                    internalListener.onFailure(exception);
+                                                }));
+                                        } else {
+                                            internalListener.onFailure(
+                                                new FlowFrameworkException(
+                                                    "Failed to update index: " + indexName,
+                                                    Response.Status.INTERNAL_SERVER_ERROR
+                                                )
+                                            );
+                                        }
+                                    }, exception -> {
+                                        logger.error("Failed to update index " + indexName, exception);
```

Comment:
Here and for the rest
```suggestion
                                        logger.error(() ->  new ParameterizedMessage("Failed to update index {}", indexName), exception);
```

Reply:
I (tentatively) disagree with this suggestion, at least for now.  @owaiskazi19 can you tell me what the difference here is?

Reply:
Something we used to do while working on core. A better way to write the parameterized message. Nothing specific.

Reply:
Got it. I think that may have been necessary with earlier log4j syntax but it's not any more. 

I agree the concatenation should be replaced, but I think `logger.error("Failed to update index {}", indexName, exception);` should produce exactly the same thing.